src/indexer/index_writer.rs

```diff
@@ -643,12 +647,45 @@ impl IndexWriter {
     pub fn add_document(&mut self, document: Document) -> u64 {
         let opstamp = self.stamper.stamp();
         let add_operation = AddOperation { opstamp, document };
-        let send_result = self.document_sender.send(add_operation);
+        let send_result = self.document_sender.send(vec![add_operation]);
         if let Err(e) = send_result {
             panic!("Failed to index document. Sending to indexing channel failed. This probably means all of the indexing threads have panicked. {:?}", e);
         }
         opstamp
     }
+
+    /// Runs a group of document operations.
+    pub fn run(&mut self, user_operations: &[UserOperation]) -> Option<u64> {
+        let mut stamps = self.stamper.stamps(user_operations.len() as u64);
```

Comment:
Opstamp are free, I suggest allocation `user_operations.len() + 1`  operations and give the last one to the "batch".

That way you can change the `Option<u64>`  to `u64`. (Even empty operations return an opstamp).
There might be value (maybe?) for the batch and the last operation to have a different opstamp too.

I also allocate opstamp to operations like `commit` etc. I suspect it might help people not having to think too much about the `n` vs `n+1` problem. 

If you choose this path, you can set last opstamp to the last opstamp of the range directly and `.zip(...)` the iterators.

Reply:
`There might be value (maybe?) for the batch and the last operation to have a different opstamp too.`
I think so. This "extra" opstamp could give delimitation to a group of operations. A sort of forensics? 

I'll make the changes.